index.js

```diff
@@ -134,14 +134,21 @@ function fireChangeEvent(target, result) {
   for (const key in result) {
     const value = Math.round(result[key] * ratio)
     result[key] = value
-    const slottedInput = target.querySelector(`[data-image-crop-input='${key}']`)
-    if (slottedInput) slottedInput.value = value
+    target.updateValues(result)
   }
 
   target.dispatchEvent(new CustomEvent('image-crop-change', {bubbles: true, detail: result}))
 }
 
+function submit(event) {
+  if (event.key === 'Enter' && event.target.form) event.target.form.submit()
+}
+
 export class ImageCropElement extends HTMLElement {
+  static formAssociated = true
+
+  #internals = this.attachInternals()
```

Comment:
Are the private attributes a prerequisite to using this API?

Reply:
I don't believe they are. I think this can be done with a `WeakMap`